lib/src/coap_client.dart

```diff
@@ -131,7 +132,7 @@ class CoapClient {
     final String path, {
     final CoapMediaType? accept,
     final bool confirmable = true,
-    final List<CoapOption>? options,
+    final List<Option<dynamic>>? options,
```

Comment:
Could we use `Object` rather than `dynamic`? It's a little more specific.

Reply:
Had to use `Object?` because of the Empty option format, hope it's still an improvement.